src/vs/workbench/contrib/chat/browser/chatWidget.ts

```diff
@@ -460,6 +471,11 @@ export class ChatWidget extends Disposable implements IChatWidget {
 		return this.inputPart.hasFocus();
 	}
 
+	refreshParsedInput() {
+		this.parsedChatRequest = this.instantiationService.createInstance(ChatRequestParser).parseChatRequest(this.viewModel!.sessionId, this.getInput(), this.location, { selectedAgent: this._lastSelectedAgent });
+		this._onDidChangeParsedInput.fire();
```

Comment:
Where is this being listened to in your scenario? I think I see the problem but I don't really like this pattern of what might look like triggering this from a random subset of the places that lead to changes in the result.

Reply:
https://github.com/microsoft/vscode/blob/63e5a64a61d75b5f2f26806939ef9c853881d729/src/vs/workbench/contrib/chat/browser/chatWidget.ts#L277-L279

Reply:
And you basically just want to know when a `#file` showed up in the query? Would it suffice to have some other event to listen to when that changes?

Reply:
Or when a `#file` disappears from the query

Reply:
Right, maybe you can have an event on `ChatDynamicVariableModel` to listen to, or maybe these should live in the attachment model somehow. Adding an item to my debt list.